yarn-project/telemetry-client/src/telemetry.ts

```diff
@@ -221,3 +223,45 @@ export function wrapCallbackInSpan<F extends (...args: any[]) => any>(
     }
   }) as F;
 }
+
+export function runInSpan<A extends any[], R extends any>(
```

Comment:
Should we merge this method's logic with `trackSpan`'s?

Reply:
Yes, I think they're good candidates to merge together (the only difference is the span is activated in the new function). I'll log an issue for this